### ISSUE TITLE
[nova-network] RPC v4.1.2 and v4.1.3 Quantum: Failed to create OVS patch port

### ISSUE THREAD

**Issue by jameswthorne** [rcbops/chef-cookbooks#702]
This issue was originally fixed by installing **openvswitch-datapath-lts-raring-dkms** on the network and compute nodes as detailed by the following issue: https://github.com/rcbops/chef-cookbooks/issues/598

However, this change was reverted and now fresh RPC v4.1.2 or v4.1.3 environments using Quantum have Quantum communication issues between the network and compute nodes.

My _exact_ build steps for an RPC v4.1.2 or v4.1.3 environment using Quantum can be found at http://public.thornelabs.net/rpcv413-ubuntu-quantum.txt

If you follow those steps the following error will be echoed repeatedly in **/var/log/quantum/openvswitch-agent.log** on the network and compute nodes:

```
2013-12-08 23:11:34    ERROR [quantum.plugins.openvswitch.agent.ovs_quantum_agent] Failed 
to create OVS patch port. Cannot have tunneling enabled on this agent, since this version 
of OVS does not support tunnels or patch ports. Agent terminated!
```

The Vagrant Base Boxes are running Ubuntu Server LTS 12.04.3 with kernel 3.8.0-29-generic. The Open vSwitch package versions can be found below:

```
root@network1:/var/log/quantum# dpkg -l | grep openvswitch
ii  openvswitch-common               1.4.0-1ubuntu1.5             Open vSwitch common components
ii  openvswitch-datapath-dkms        1.4.0-1ubuntu1.5             Open vSwitch datapath module source - DKMS version
ii  openvswitch-switch               1.4.0-1ubuntu1.5             Open vSwitch switch implementations
ii  quantum-plugin-openvswitch       1:2013.1.3-0ubuntu1~cloud0   Quantum is a virtual network service for Openstack - Open vSwitch plugin
ii  quantum-plugin-openvswitch-agent 1:2013.1.3-0ubuntu1~cloud0   Quantum is a virtual network service for Openstack - Open vSwitch plugin agent
```

**Comment by Apsu** [comment]
We've seen this issue before, and we're pretty sure we know the cause and solution, as well as why this broke with the switch to "openvswitch-datapath-dkms" in v4.1.3.

Namely: The grizzly target in cloud-archive doesn't have the newer openvswitch-datapath-dkms (1.10), or even the package at all, which means it falls back to the main repos and installs 1.4, as indicated in the package list above. This version was known to be broken on 3.8 kernels, which is why we originally moved to the openvswitch-datapath-lts-raring-dkms package, but with the 3.8.0-34 revision that dropped into precise last week, _that_ package is now broken in a much more horrible way, causing kernel panics.

So, we're going to resolve the issue by getting the havana cloud-archive openvswitch-datapath-dkms packages into our repo, after I finish verifying that they work against some more revisions of 3.8; we had already tested 3.2, 3.5 and 3.8.0-33/34. I'll update this once we have some more information.

**Comment by jameswthorne** [comment]
Good to know. Thanks for the update Evan.

**Comment by rpawlik** [comment]
Has this been resolved? We have a customer with a newly installed compute node running kernel 3.8.0-35.

```
2014-01-09 17:14:59    ERROR [quantum.plugins.openvswitch.agent.ovs_quantum_agent] Failed to create OVS patch port. Cannot have tunneling enabled on this agent, since this version of OVS does not support tunnels or patch ports. Agent terminated!
```

```
root@compute04:~# uname -r
3.8.0-34-generic
```

```
root@compute04:~# dpkg -l | grep openvswitch
ii  openvswitch-common               1.4.0-1ubuntu1.5                            Open vSwitch common components
ii  openvswitch-datapath-dkms        1.4.0-1ubuntu1.5                            Open vSwitch datapath module source - DKMS version
ii  openvswitch-switch               1.4.0-1ubuntu1.5                            Open vSwitch switch implementations
ii  quantum-plugin-openvswitch       1:2013.1.4-0ubuntu1~cloud0                  Quantum is a virtual network service for Openstack - Open vSwitch plugin
ii  quantum-plugin-openvswitch-agent 1:2013.1.4-0ubuntu1~cloud0                  Quantum is a virtual network service for Openstack - Open vSwitch plugin agent
```

**Comment by jameswthorne** [comment]
It has been in RPC v4.1.3. I can't speak for RPC v4.1.2.

**Comment by rpawlik** [comment]
Thanks, James. 

Do you guys happen to know what the fix is in case this customer doesn't want to upgrade? Is it installing the openvswitch-dkms package from the Havana repo?

**Comment by Apsu** [comment]
This has been resolved in 4.1.4rc/4.2.2rc and will be in the upcoming releases.

**Comment by jacobwagner** [comment]
Verified by checking for package version, hence closing.

```
root@qa-precise-pool104:~# dpkg -l | grep openvswitch
ii  neutron-plugin-openvswitch           1:2013.2.1-0ubuntu1~cloud0                          Neutron is a virtual network service for Openstack - Open vSwitch plugin
ii  neutron-plugin-openvswitch-agent     1:2013.2.1-0ubuntu1~cloud0                          Neutron is a virtual network service for Openstack - Open vSwitch plugin agent
ii  openvswitch-common                   1.10.2-0ubuntu2~cloud0                              Open vSwitch common components
ii  openvswitch-datapath-dkms            1.10.2-0ubuntu2~cloud0                              Open vSwitch datapath module source - DKMS version
ii  openvswitch-switch                   1.10.2-0ubuntu2~cloud0                              Open vSwitch switch implementations
```
